Makefile

```
@@ -107,7 +107,7 @@ check-code-security:
 .PHONY: check-format
 check-format:
 	mix format --dry-run --check-formatted
-	npx prettier --ignore-path assets/.prettierignore --check $(PRETTIER_FILES_PATTERN)
+	assets/node_modules/.bin/prettier --ignore-path assets/.prettierignore --check $(PRETTIER_FILES_PATTERN)
```

Why not using `cd` to change the directory before executing the script?

```suggestion
	cd assets && npx prettier --ignore-path .prettierignore --check $(PRETTIER_FILES_PATTERN)

---

And in fact, we could also remove the `--ignore-path` flag since we would already be in the right folder! 🤯

---

to avoid `cd` messing with the user's current folder: 
`(cd assets; npx prettier --ignore-path .prettierignore --check $(PRETTIER_FILES_PATTERN))`

---

Makefiles aren’t shell scripts, they execute in a separate shell and therefore don’t mess with the current directory in which you execute the script

---

You’re right, it wouldn’t mess with the user path, but I thought best to avoid it to prevent this:

```make
format:
	mix format                          # Working directory is `.`
	cd assets && npx prettier --write … # Working directory is `./assets`
    other-formatting-tool               # Working directory is still `./assets`
```

I guess we could use:

```make
cd assets && npx prettier --write … && cd ..
```

Even if `npx prettier --write …` fails and the second `&&` is skipped, it wouldn’t matter because the `other-formatting-tool` wouldn’t get executed anyway.

What do you guys think?

---

In fact each line is executed separately, I tested it and given a target like this:

```
.PHONY: lol
lol:
	cd app && pwd
	pwd
```

I have the output:

```
❯ /Volumes/Mirego/projects/my-project
❯ make lol
cd app && pwd
/Volumes/Mirego/projects/my-project/app
pwd
/Volumes/Mirego/projects/my-project
```

---

Oh, awesome then. Reverting to `cd assets && npx …`! 🎉